### PR TITLE
Enable Metal performance HUD

### PR DESCRIPTION
**Pull request by elijah-semyonov**
## Proposed Changes

If `Info.plist` contains `MetalHudEnabled` set to true, configure CAMetalLayer to show performance hud.
Update availability check logic to use `available` API from skiko in ComposeWindow.

## Testing

Test: N/A

IMG_3A8D66B7D2B9-1

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt [hunk 1]
```diff
@@ -345,6 +356,23 @@ internal class MetalRedrawer(
             }
         }
     }
+
+    private fun configureMetalHUDIfNeeded() {
+        if (available(OS.Ios to OSVersion(16))) {
+            val entry = NSBundle
+                .mainBundle
+                .objectForInfoDictionaryKey("MetalHudEnabled") as? NSNumber
```
Comment (igordmn) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt:364]: Is it a custom property? Will the name "Compose.MetalHudEnabled" be by the naming style?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt [hunk 2]
```diff
@@ -345,6 +356,23 @@ internal class MetalRedrawer(
...
+
+            if (entry?.boolValue == true) {
+                dispatch_async(dispatch_get_main_queue()) {
```
Comment (igordmn) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt]: Do we need to dispatch? The constructor should be called in the main thread

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt (hunk 1 above)
Comment (elijah-semyonov) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt:364]: No, it's the actual plist value required by Apple. User can add this to plist, no API change in ComposeConfiguration is needed.

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt (hunk 2 above)
Comment (elijah-semyonov) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt]: Oof, that's a copy-paste error. Thanks. Removed.

**Review by igordmn — APPROVED** [review]

**Pull request merged**